api/envoy/api/v2/core/protocol.proto

```diff
@@ -27,6 +27,14 @@ message HttpProtocolOptions {
   // Note that request based timeouts mean that HTTP/2 PINGs will not keep the connection alive.
   google.protobuf.Duration idle_timeout = 1;
 
+  // The maximum lifetime of downstream connection. The lifetime timeout is defined
```

Comment:
I think this should just be "of a connection", and you can then qualify that it only works for downstream. There's no reason this can't be implemented to apply to upstream when these protocol options are used by cluster manager.

Reply:
Updated comment. What would be the best way add a note that it's implemented for downstream connections only?

Reply:
Arguably an API shephard thing but I'd prefer we either have folks sign up for the complete functionality (if we think we want both) in which case a TODO and follow-up PR is OK, or we move this to the HCM with the understanding we're only doing it for downstream.

Reply:
moved to HttpProtocolOptions